docs/docs/1.2.3/policies/virtual-outbound.md

```diff
@@ -0,0 +1,254 @@
+
+Possible use cases are:
+
+1) Preserving hostnames when migrating to service mesh (This enables migration without modifying app configuration).
+2) Providing multiple hostnames for reaching the same service (when renaming or for usability).
+3) Providing specific routes (reach a specific pod in a service when using StatefulSets, add an url to reach a specific version of a service).
+4) Expose multiple inbounds on different ports.
+
+## Usage
+
+`conf.host` and `conf.port` are processed as [go text templates](https://pkg.go.dev/text/template) with a map of key, value constituted from `conf.parameters`.
+
+For example a dataplane with this definition:
+
+```yaml
+type: Dataplane
+mesh: default
+name: backend-1
+networking:
+  address: 192.168.0.2
+inbound:
+  - port: 9000
+    servicePort: 6379
+    tags:
+      kuma.io/service: backend
+      version: v1
+      port: 1800
+```
+
+A virtual outbound with definition:
+
+```yaml
+type: VirtualOutbound
+mesh: default
+name: test
+selectors:
+  - match:
+      kuma.io/service: "*"
+conf:
+  host: "{{.v}}.{{.service}}.mesh"
+  port: "{{.port}}"
+  parameters:
+    - name: service
+      tagKey: "kuma.io/service"
+    - name: port
+      tagKey: port
+    - name: v
+      tagKey: version
+```
+
+Will create a hostname: `v1.backend.mesh` and port: `1800`.
+
+Virtual Outbounds have some constraints/limitations:
+
+- It only works when using transparent proxy.
+- When not using the [data plane proxy DNS](/docs/1.2.3/networking/dns.md#data-plane-proxy-dns), all generated hostnames must end with the value of the configuration `dns_server.domain` (whose default is `.mesh`).
+- If a `tagKey` is absent it uses `name` as `tagKey`.
+- `name` must be alphanumeric as it's used as a go template key.
+- for a virtual-outbound set of parameters all `name` must be unique.
+- `kuma.io/service` must be in the parameters even if it's unused in the template (this prevents defining hostnames that spans services).
+
+For each virtual outbound the Kuma control plane will process all dataplanes matching the selector.
+It will then apply the templates for `conf.host` and `conf.port` and assign a virtual ip for each unique set defined by all the `tagKeys` value in the parameters.
+This means that different hostnames may resolve to the same vip because they map to the same tag set.
+
+### Collisions
+
+We rely on user defined templates it is thus possible to have hostname collisions.
+
+We apply these rules to avoid them:
+
+1) For collisions between virtual outbounds, the virtual outbound with the highest priority takes over as [explained in the policy doc](../../1.2.2/policies/how-kuma-chooses-the-right-policy-to-apply.md).
+2) For all collisions log messages will be emitted. 
+3) Because only service tags are propagated across zones complex virtual outbounds will not work for cross-zone traffic. 
```

Comment:
```suggestion
When duplicate hostnames are detected, the virtual outbound with the highest priority takes over. For more information, see [the documentation on ow Kuma chooses the right poliy](../../1.2.2/policies/how-kuma-chooses-the-right-policy-to-apply.md). All duplicate instances are logged. 
```

n.b content removed from here is put further up the page bc it's better to provide that kind of warning early in the doc :D